independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/QuarkusBootstrap.java

```diff
@@ -168,7 +170,7 @@ public CuratedApplication bootstrap() throws BootstrapException {
                 appModelFactory.setEnableClasspathCache(true);
             }
         }
-        return new CuratedApplication(this, appModelFactory.resolveAppModel(), classLoadingConfig);
+        return new CuratedApplication(this, appModelFactory.resolveAppModel(), classLoadingConfig, assertionsEnabled);
```

Comment:
You don't need to pass this in as a constructor argument, as it has access to QuarkusBootstrap.